crates/ratchet-core/src/ops/cache.rs

```diff
@@ -151,7 +151,12 @@ impl MetaOperation for Cache {
     }
 
     fn kernel_element(&self, _dst: &Tensor) -> KernelElement {
-        KernelElement::Scalar
+        let numel = self.input.shape().numel();
+        if numel % 4 == 0 {
```

Comment:
No reason it should be just Vec4!

Reply:
Understood. Got carried away last time.

Adding conditions for `Vec2`, `Vec4`, and `Scalar`, just like in many other ops.